### PR TITLE
fix: LEAP-1352: API for Bulk Annotation Configuration Validation

### PR DESCRIPTION
**Pull request by triklozoid**
- add a new property `config_suitable_for_bulk_annotation` into /api/projects/ID API. it validates that label config supports bulk annotation and return the result

### CODE REVIEW

File: label_studio/projects/serializers.py [hunk 1]
```diff
@@ -82,6 +98,46 @@ def user_id(self):
...
+            ParagraphLabelsTag,
+            TimeSeriesLabelsTag,
+        ]
+
+        # Return False if any disallowed tag is present
+        for tag_class in disallowed_tags:
+            if li.find_tags_by_class(tag_class):
+                return False
+
+        # For all ChoicesTag and TaxonomyTag, check that perRegion and perItem are not enabled
```
Comment (hlomzik) [inline label_studio/projects/serializers.py]: I think, we should also check that `Taxonomy` doesn't have `apiUrl` param

File: label_studio/projects/serializers.py [hunk 2]
```diff
@@ -82,6 +98,46 @@ def user_id(self):
...
+        # List of tags that should not be present
+        disallowed_tags = [
+            LabelsTag,
+            EllipseLabelsTag,
+            KeyPointLabelsTag,
+            PolygonLabelsTag,
+            RectangleLabelsTag,
+            HyperTextLabelsTag,
+            ParagraphLabelsTag,
+            TimeSeriesLabelsTag,
```
Comment (hlomzik) [inline label_studio/projects/serializers.py]: `BrushLabels` are missing + all usual image region tags should also be a **no**: `Brush`, `Ellipse`, `KeyPoint`, `Polygon`, `Rectangle`
```suggestion
            LabelsTag,
            BrushTag,
            BrushLabelsTag,
            EllipseTag,
            EllipseLabelsTag,
            KeyPointTag,
            KeyPointLabelsTag,
            PolygonTag,
            PolygonLabelsTag,
            RectangleTag,
            RectangleLabelsTag,
            HyperTextLabelsTag,
            ParagraphLabelsTag,
            TimeSeriesLabelsTag,
```

**Review by hlomzik — CHANGES_REQUESTED** [review]

File: label_studio/projects/serializers.py (hunk 1 above)
Comment (hlomzik) [inline label_studio/projects/serializers.py]: All classification tags should be checked: Choices, DateTime, Number, Rating, TextArea

File: label_studio/projects/serializers.py (hunk 2 above)
Comment (Gondragos) [reply label_studio/projects/serializers.py]: Plus maybe `MagicWand`

File: label_studio/projects/serializers.py (hunk 1 above)
Comment (Gondragos) [reply label_studio/projects/serializers.py]: And most probably `value` in `Taxonomy`/`Choices` is "no" as well

File: label_studio/projects/serializers.py (hunk 2 above)
Comment (Gondragos) [reply label_studio/projects/serializers.py]: @hlomzik Should we do smthng with `VideoRectangle`, `TimelineLabels`?

Comment (hlomzik) [reply label_studio/projects/serializers.py]: Oh, yes, thank you! All these 3 tags should also be banned =)

**Review by jombooth — APPROVED** [review]
API implementation LGTM but of course this shouldn't be merged without addressing all feedback from @Gondragos and @hlomzik

File: label_studio/projects/serializers.py (hunk 2 above)
Comment (triklozoid) [reply label_studio/projects/serializers.py]: I added VideoRectangle, unfortunately, TimelineLabels and MagicWand is not supported by our parsing library yet, we can add them in a follow-up ticket.

File: label_studio/projects/serializers.py (hunk 1 above)
Comment (triklozoid) [reply label_studio/projects/serializers.py]: > And most probably value in Taxonomy/Choices is "no" as well

@Gondragos  I didn't understand this. 
We don't have value in Taxonomy tag for our default config example. Does it mean, that we don't support bulk annotation for it? 

```
<View>
  <Text name="text" value="$text"/>
  <Taxonomy name="taxonomy" toName="text">
    <Choice value="Archaea" />
    <Choice value="Bacteria" />
    <Choice value="Eukarya">
      <Choice value="Human" />
      <Choice value="Oppossum" />
      <Choice value="Extraterrestial" />
    </Choice>
  </Taxonomy>
</View>
```

**Review by hlomzik — APPROVED** [review]

**Comment by MihajloHoma** [comment]
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/12297200695)
> Successfully merged:  create mode 100644 docs/source/guide/upgrade_community.md

**Review by MihajloHoma — APPROVED** [review]

**Comment by triklozoid** [comment]
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/12315420366)
> Successfully merged:  1 file changed, 1 insertion(+), 1 deletion(-)

**Pull request merged**